src/dbnode/persist/fs/commitlog/commit_log.go

```diff
@@ -61,30 +61,45 @@ type commitLogFailFn func(err error)
 type completionFn func(err error)
 
 type commitLog struct {
-	sync.RWMutex
+	flushState  flushState
+	writerState writerState
+	closedState closedState
+	// Associated with the closedState, but stored separately since
+	// it does not require the closedState lock to be acquired before
+	// being accessed.
+	closeErr chan error
+
+	// TODO(r): replace buffered channel with concurrent striped
+	// circular buffer to avoid central write lock contention.
+	writes chan commitLogWrite
+
 	opts  Options
 	nowFn clock.NowFn
-
-	log xlog.Logger
+	log   xlog.Logger
 
 	newCommitLogWriterFn newCommitLogWriterFn
 	writeFn              writeCommitLogFn
 	commitLogFailFn      commitLogFailFn
-	writer               commitLogWriter
 
-	// TODO(r): replace buffered channel with concurrent striped
-	// circular buffer to avoid central write lock contention
-	writes chan commitLogWrite
+	metrics commitLogMetrics
+}
 
-	flushMutex      sync.RWMutex
+type flushState struct {
+	sync.RWMutex
 	lastFlushAt     time.Time
 	pendingFlushFns []completionFn
+}
 
+type writerState struct {
+	sync.RWMutex
+	writer         commitLogWriter
 	writerExpireAt time.Time
-	closed         bool
-	closeErr       chan error
+	activeFile     *File
+}
 
-	metrics commitLogMetrics
+type closedState struct {
+	sync.RWMutex
+	closed bool
```

Comment:
could you replace the `closedState` with an atomic. It'll make a lot of the code simpler